### DISCUSSION TITLE
Docker Tasks could be more generic to support all API methods.

### DISCUSSION THREAD

**Discussion by bukowa** [PrefectHQ/prefect#3003] [Ideas] [answered]
Docker tasks are limited, and we cannot pass custom kwargs into new APIClient. Do you think that Docker based Tasks could be more generic / abstract to handle all API endpoints?

**Comment by joshmeek** [comment, marked as answer]
@bukowa Yeah these could definitely be expanded upon! For example, the `CreateContainer` task has `extra_docker_kwargs` for setting any custom kwargs that you want to forward to the Docker client:

https://github.com/PrefectHQ/prefect/blob/ce234bf92bd42ebea6b261ba7c2f8213ea6c49b3/src/prefect/tasks/docker/containers.py#L27

The task library is generally community-driven and we highly encourage contributions! 🙂

**Comment by bukowa** [comment]
What about something like that?
@joshmeek 
````python
import operator

import docker
from prefect import Task


class DockerTask(Task):

    def __init__(self, client, method, **kwargs):
        self.docker_func = operator.attrgetter(method)(client)
        super().__init__(**kwargs)

    def run(self, **kwargs) -> None:
        return self.docker_func(**kwargs)


if __name__ == '__main__':
    client = docker.client.APIClient()
    task = DockerTask(
        client=client,
        method='containers',
    )
    task.run(all=True)
````

**Reply by joshmeek** [reply]
Yeah that would be an appropriate docker task!
